src/autocomplete.js

```diff
@@ -267,43 +396,48 @@ class Autocomplete extends Component {
     document.body.addEventListener('mousedown', onMouseDown)
     document.body.addEventListener('mouseup', onMouseUp)
 
-    this.unregisterClickTracker = () => {
+    this.cleanup = () => {
+      this._isMounted = false
       document.body.removeEventListener('mousedown', onMouseDown)
       document.body.removeEventListener('mouseup', onMouseUp)
     }
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      prevState.highlightedIndex !== this.state.highlightedIndex ||
+      this.state.value !== prevState.value
+    ) {
+      this.updateStatus()
+    }
+  }
+
   componentWillUnmount() {
-    this.unregisterClickTracker() // avoids memory leak
+    this.cleanup() // avoids memory leak
   }
 
   render() {
+    // because the items are rerendered every time we call the children
+    // we clear this out each render and
+    this.items = []
     const {
-      component: AutocompleteComponent,
       children,
       // eslint-disable-next-line no-unused-vars
-      defaultHighlightedIndex,
-      getA11yStatusMessage,
+      defaultSelectedItem,
       // eslint-disable-next-line no-unused-vars
       getValue,
       // eslint-disable-next-line no-unused-vars
-      onChange,
+      getA11yStatusMessage,
       // eslint-disable-next-line no-unused-vars
-      innerRef,
+      defaultHighlightedIndex,
+      // eslint-disable-next-line no-unused-vars
+      onChange,
       ...rest
     } = this.props
     return (
-      <AutocompleteComponent ref={this.ref} {...rest}>
-        {children}
-        <Status
-          getA11yStatusMessage={getA11yStatusMessage}
-          getItemFromIndex={this.getItemFromIndex}
-          getValue={this.getValue}
-          highlightedIndex={this.state.highlightedIndex}
-          resultCount={this.items.length}
-          value={this.state.value}
-        />
-      </AutocompleteComponent>
+      <div {...rest} ref={this.rootRef} onClick={this.handleClick}>
```

Comment:
Well, the problem is that if we don't render a div there's no reasonable way for us to find the `_rootNode` for some of the mousedown/up handlers we have. I think that rendering a single div is a nice tradeoff. What else would you render?